examples/datalog/merge_intervals.flix

```diff
@@ -0,0 +1,22 @@
+// https://leetcode.com/problems/merge-intervals/
+// Given a list of intervals where intervals[i] = [start_i, end_i], merge all overlapping intervals, and
+// return an array of the non-overlapping intervals that cover all the intervals in the input.
+
+def merge(intervals: List[(Int32, Int32)]): Array[(Int32, Int32)] & Pure =
```

Comment:
@magnus-madsen @mlutze Is it possible to "project" with an Array? I couldn't get it working.

Reply:
Arrays are not `Foldable`, so no.